crates/protocol/interop/src/safety.rs

```diff
@@ -19,6 +21,27 @@ pub enum SafetyLevel {
     Invalid,
 }
 
+impl FromStr for SafetyLevel {
+    type Err = SafetyLevelParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "finalized" => Ok(Self::Finalized),
+            "safe" => Ok(Self::Safe),
+            "local_safe" | "localsafe" => Ok(Self::LocalSafe),
+            "cross_unsafe" | "crossunsafe" => Ok(Self::CrossUnsafe),
```

Comment:
don't think these underscore `_` values are accurate, since this uses kebab-case we should also support those or replace the `_` with `-`